### ISSUE TITLE
Improve error messages

### ISSUE THREAD

**Issue by rgoldberg** [mas-cli/mas#533]
## Your Environment

- `mas version`: 1.8.6
- macOS version (`system_profiler SPSoftwareDataType -detailLevel mini`):
Software:

    System Software Overview:

      System Version: macOS 12.7.6 (21H1320)
      Kernel Version: Darwin 21.6.0
      Time since boot: 1 day 14:20
### mas Install Method

- [x] `brew install mas` (homebrew-core)
- [] [mas-cli/tap](https://github.com/mas-cli/homebrew-tap)
- [] `.pkg` installer from [releases](https://github.com/mas-cli/mas/releases)
- [] Built from source
  - Fork/branch: ? (e.g. mas-cli/main)
  - Xcode version: 10.?

## Describe the Bug

When invalid arguments are supplied to `mas` commands, many output cryptic and/or uninformative errors.

Can either create a new issue for each output issue, or can just try all commands one day to find & fix all uninformative error message, and to standardized their outputs, whatever you want.

## To Reproduce

1. ```mas install 2143935391```

## Expected Behavior
An error like:

```
Unknown app ID: 2143935391
```

## Actual Behavior

```
Error: Download request failed: The server gave an error during download: 500 Internal Server Error.
```

## Screenshots, Terminal Output

See above

**Comment by Austin1** [comment]
I found this issue when searching for “unknown app id” in the github issues log. I’m getting this error and this issue seems to insinuate that I should be getting a more detailed error message? Here is what I’m doing:

```
austin@austins-mac-mini ~ % mas search channelsdvr
...
  1405359767  Channels: Whole Home DVR        (6.1.2)
...
austin@austins-mac-mini ~ % mas purchase 1405359767
Error: Unknown app ID 1405359767
austin@austins-mac-mini ~ % mas install 1405359767
Error: Unknown app ID 1405359767
```

**Comment by rgoldberg** [comment]
@Austin1 I'm getting:

```console
$ mas search channelsdvr
Error: No apps found
```

What region is your Mac App Store using? What region is your macOS user using (from System Settings)? (mine are both US)

`search` & `info` use the macOS user region to pass to the public Apple iTunes Search Web API, while `purchase` & `install` use private Apple Frameworks that internally use the Mac App Store region.

We'd prefer to use the Mac App Store region for everything, but the people who investigated this a long time ago couldn't find the Mac App Store region programmatically, so they used the macOS user region, as they are normally the same. We hope to [find Mac App Store region programmatic access](https://github.com/mas-cli/mas/issues/75#issuecomment-2351740041) later.

When the regions are different, you can get different results from different commands.

**Comment by rgoldberg** [comment]
@Austin1 I will soon submit a PR to use the Mac App Store region instead of the macOS region. The issue is #684. Please subscribe to that.

The PR will tentatively be release in 1.8.9. Not sure exactly when that will be released.

if you want to test if that PR will fix your problem after it is merged into main but before 1.8.9 is released, you can run:

```shell
brew install --HEAD mas
```
